### PR TITLE
Add a search backend specific css class to results list.

### PR DESCRIPTION
**Pull request by EreMaijala**
This makes it possible to style a specific results list without having to resort to the template-dir-.. or template-name-.. classes.

Extracted from #2341.

### CODE REVIEW

File: themes/bootstrap3/templates/search/results.phtml
Comment (crhallberg) [inline themes/bootstrap3/templates/search/results.phtml]: I like the addition of this class a lot! This regex produces good classnames too, with maybe the exception of EDS:

- EDS: e-d-s
- Solr: solr
- SolrWeb: solr-web
- Summon: summon

Comment (crhallberg) [reply themes/bootstrap3/templates/search/results.phtml]: Also, do we want a prefix on this class? `search-results-solr`?

File: themes/bootstrap3/templates/search/list-grid.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/search/list-grid.phtml]: Should we do something like:

`class="<?=$this->escapeHtmlAttr(trim('search-grid ' . ($this->resultsClass ?? ''))?>"`

...to avoid empty whitespace at the end of the HTML attribute?

File: themes/bootstrap3/templates/search/list-list.phtml
Comment (demiankatz) [inline themes/bootstrap3/templates/search/list-list.phtml]: Same whitespace comment as above.

File: themes/bootstrap3/templates/search/results.phtml
Comment (EreMaijala) [reply themes/bootstrap3/templates/search/results.phtml]: Oops, EDS wasn't great. I've added a special case for all-caps names. Also prefixed now.

File: themes/bootstrap3/templates/search/list-grid.phtml
Comment (EreMaijala) [reply themes/bootstrap3/templates/search/list-grid.phtml]: Fixed.

File: themes/bootstrap3/templates/search/list-list.phtml
Comment (EreMaijala) [reply themes/bootstrap3/templates/search/list-list.phtml]: Fixed.

**Review by demiankatz — APPROVED** [review]
Thanks for the updates, @EreMaijala, this all looks good to me now... but I'll wait for @crhallberg's approval before merging in case he has any last-minute thoughts.

**Comment by crhallberg** [comment]
Looks good to me!

**Comment by demiankatz** [comment]
Thanks, @crhallberg -- merging now!

**Pull request merged**
